### ISSUE TITLE
Companion crashes when opening model or simulate if SD-Path in application settings not empty

### ISSUE THREAD

**Issue by mamaretti** [EdgeTX/edgetx#1961]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Companion

### Current Behavior

when I try to start simulator or try to edit a model Companion crashes (2.5, 2.6, 2.7....)

but only when entered a correct SD-path in App-settings (settings->radio-profile->other settings)

if the field is empty  simulator and edit a model is working.

i'm on macOS Mojave

### Expected Behavior

no crash

### Steps To Reproduce

open document.etx file an try to start simulator.
check sd-path settings before.

### Version

2.6.0

### Transmitter

Radiomaster TX16S

### Anything else?
<img width="933" alt="Bildschirmfoto 2022-05-13 um 17 29 21" src="https://user-images.githubusercontent.com/3448514/168316952-f48d4143-b6ff-46aa-b5fe-8801a3c7b50b.png">

**Comment by mamaretti** [comment]
solved by myself.
it crashes with custom theme.
i created a new theme and now its working.
similar issue https://github.com/EdgeTX/edgetx/issues/1517#issuecomment-1025404356

**Comment by elecpower** [comment]
> it crashes with custom theme.
> i created a new theme and now its working.

@mamaretti I have been trying to replicate and cannot reproduce so I am obviously not taking the same steps. Could you please explain further. Thanks.

**Comment by mamaretti** [comment]
hi elepower.
on EdgeTX 2.6 I created a custom theme some time ago.
on the Radio the theme ist working without problems.

companion can read the SD-Data by entering the correct path  to system location.

what I have done: I deleted my theme and  selectedtheme.txt 
so that EdgeTX and companion load the default EdgeTX-Theme. 

after restart companion simulator and modell-editor works again correctly.

then I created I a new custom theme via simulator. and now companion, modell-editor and radio is working with new theme.

I compared the both "theme.yml" files. and the only difference was that the "corrupted one" has a DASH "-" in the info-attribute of summary. perhaps this was the problem.

**Comment by mamaretti** [comment]
currupt , old theme (?)

`---
summary:
  name: Blue
  author: Me
  info:

colors:
  PRIMARY1:   0x000000
  PRIMARY2:   0xF8FCF8
  PRIMARY3:   0x0074B0
  SECONDARY1: 0x282C28
  SECONDARY2: 0xC8E8F8
  SECONDARY3: 0x98C0D8
  FOCUS:      0x283880
  EDIT:       0x4074E8
  ACTIVE:     0xE0EC90
  WARNING:    0xE00000
  DISABLED:   0x888C88`

ok:

`---
summary:
  name: blue2
  author: me
  info: 

colors:
  PRIMARY1:   0x000000
  PRIMARY2:   0xF8FCF8
  PRIMARY3:   0x0074B0
  SECONDARY1: 0x282C28
  SECONDARY2: 0xC8E8F8
  SECONDARY3: 0x98C0D8
  FOCUS:      0x283880
  EDIT:       0x4074E8
  ACTIVE:     0xE0EC90
  WARNING:    0xE00000
  DISABLED:   0x888C88
`

**Comment by elecpower** [comment]
@mamaretti thank you for the info. The dash shouldn't crash Companion  but hey stranger things have happened. I'll experiment so more.

**Comment by mamaretti** [comment]
I did a file compare an the only difference was the dash and the name,
if you want I can send you my SD-Backups

**Comment by elecpower** [comment]
The dash may not be the dash character as it does not display in the email copy of your post. Could you please upload the suspect theme.yml file as a zip file so I can check the file in a hex editor

**Comment by mamaretti** [comment]
attached the hole theme-folder with the non-working-theme
[THEMES.zip](https://github.com/EdgeTX/edgetx/files/8698140/THEMES.zip)

**Comment by elecpower** [comment]
Thanks

**Comment by elecpower** [comment]
Your theme generates the following

> terminate called after throwing an instance of 'YAML::ParserException'
  what():  yaml-cpp: error at line 5, column 9: illegal block entry

Confirms cause is the dash/hyphen. If the hyphen is enclosed in quotes then no crash as treated as text.

**Comment by elecpower** [comment]
@mamaretti thank you for your assistance

**Comment by ZZ-Cat** [comment]
I have experienced something similar to this.
I first noticed it in Companion 2.7.1.
I rolled back all the way to 2.6.0 & did a completely fresh install of Companion 2.6.0.
I left the SD Structure Path & Backup Folder fields blank; loaded up a new file; created a new model; & everything worked fine.

The very second I filled out the SD Structure Path & Backup Folder fields with where I normally put them (in my One Drive), sure enough... opening the same blank model crashed Companion.

The interesting part is... I recently updated Windows 10 earlier in the week. Before that update, Companion has _never_ crashed on me. Not even once. Perhaps the new update may have changed the way file access permissions are handled? 🤔
